addon/computeds/format.js

```diff
@@ -1,36 +1,47 @@
 import Ember from 'ember';
-import isDescriptor from '../utils/is-descriptor';
+import emberComputed from 'ember-new-computed';
 import moment from 'moment';
+import isDescriptor from '../utils/is-descriptor';
 
-const { computed, get, assert } = Ember;
+const { get, assert } = Ember;
 const a_slice = Array.prototype.slice;
 
-export default function computedMoment(date, outputFormat, maybeInputFormat) {
-  assert('More than one argument passed into moment computed', arguments.length > 1);
+function computedFormat(date, maybeOutputFormat, maybeInputFormat) {
+  assert('At least one datetime argument required for moment computed', arguments.length);
 
-  let args = a_slice.call(arguments);
+  const args = a_slice.call(arguments);
   let result;
   args.shift();
 
-  return result = computed(date, function () {
-    let momentArgs = [get(this, date)];
+  return result = emberComputed(date, {
+    get() {
+      const momentArgs = [get(this, date)];
 
-    let propertyValues = args.map((arg) => {
-      let desc = isDescriptor.call(this, arg);
-      if (desc && result._dependentKeys.indexOf(arg) === -1) {
-        result.property(arg);
-      }
+      const propertyValues = args.map((arg) => {
+        const desc = isDescriptor.call(this, arg);
+        if (desc && result._dependentKeys.indexOf(arg) === -1) {
+          result.property(arg);
+        }
 
-      return desc ? get(this, arg) : arg;
-    });
+        return desc ? get(this, arg) : arg;
+      });
 
-    outputFormat = propertyValues[0];
+      if (propertyValues.length) {
+        maybeOutputFormat = propertyValues[0];
 
-    if (propertyValues.length > 1) {
-      maybeInputFormat = propertyValues[1];
-      momentArgs.push(maybeInputFormat);
-    }
+        if (propertyValues.length > 1) {
+          maybeInputFormat = propertyValues[1];
+          momentArgs.push(maybeInputFormat);
+        }
+      }
+      else if (this.container) {
+        const config = this.container.lookupFactory('config:environment');
```

Comment:
@stefanpenner know of a better way?  The only other thing I can think of is to wrap this computed property in a factory method where the config is passed in as an argument and returns the computed property.  Similar to what I am doing with the moment-format helper: https://github.com/stefanpenner/ember-moment/pull/76/files#diff-cb0d34a4dfd51d4326dc7c0f028f14f2R3